Make sendToProcessesForEvent compile correctly when used.

b-weinstein opened the pull request:
#### 6896e90a42a4d8ba60aabdbdbed0aa275a328cf9
<pre>
Make sendToProcessesForEvent compile correctly when used.
<a href="https://bugs.webkit.org/show_bug.cgi?id=249514">https://bugs.webkit.org/show_bug.cgi?id=249514</a>
&lt;rdar://problem/102820594&gt;

Reviewed by Timothy Hatcher.

* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::sendToProcessesForEvent):

Fire the onBeforeNavigateEvent when a frame starts to load content.
<a href="https://bugs.webkit.org/show_bug.cgi?id=249514">https://bugs.webkit.org/show_bug.cgi?id=249514</a>
rdar://102820594

Reviewed by Timothy Hatcher.

When a frame begins navigation, WebKit tells all extensions about it using the following flow.
- WebFrameLoaderClient sends a message to the WebExtensionController using IPC.
- WebExtensionController iterates over all the WebExtensionContexts and:
        - Sends a message to their WebExtensionContextProxy using IPC.

The WebExtensionContextProxy then handles this call by firing the onBeforeNavigate event on the extension.

There are some loose ends here:
1) We are firing events for some pages we shouldn&apos;t
2) The tab and frame IDs that we are vending to the listener aren&apos;t correct (they are identifiers, but not ones that have been
normalized for the required context)
3) We should be passing more information in the details dictionary.

* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm:
(WebKit::WebExtensionController::didStartProvisionalLoadForFrame): Tell every extension context about the event.
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionController.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionController.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionContextProxyCocoa.mm: Copied from Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.messages.in.
(WebKit::WebExtensionContextProxy::dispatchWebNavigationOnBeforeNavigateEvent): Create a details dictionary and
fire the &quot;onBeforeNavigate&quot; event.
* Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionControllerProxyCocoa.mm:
(WebKit::WebExtensionControllerProxy::didStartProvisionalLoadForFrame): Tell the UIProcess about this.
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.h:
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.messages.in:
* Source/WebKit/WebProcess/Extensions/WebExtensionControllerProxy.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDidStartProvisionalLoad): Tell the extensions controller about this.

Canonical link: <a href="https://commits.webkit.org/258163@main">https://commits.webkit.org/258163@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1a221cd9c965b5879e80bdde64ea0ad72a898226

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101070 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10225 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34122 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110369 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11155 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1100 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/93482 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108202 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106853 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/8446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/91712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/93482 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90351 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/23109 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/93482 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/3889 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/24626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3919 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/1032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10030 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44134 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5604 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/5686 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

kiaraarose commented on Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionContextProxyCocoa.mm:
```diff
@@ -0,0 +1,65 @@
+
+using namespace WebCore;
+
+// MARK: webNavigation support
```
I've seen both `Mark:` and `Mark: - `. Is there a styling preference of using one over the other?

b-weinstein replied:
They have slightly different behaviors in Xcode's UI.

`MARK: -` puts a separator above the item, `MARK:` doesn't.

without -:
<img width="641" alt="Screenshot 2022-12-20 at 10 25 45 AM" src="https://user-images.githubusercontent.com/114437188/208739254-cc663d1b-0428-4510-ae76-3696dab8308b.png">

vs with -:
<img width="582" alt="Screenshot 2022-12-20 at 10 26 45 AM" src="https://user-images.githubusercontent.com/114437188/208739414-08f691da-918b-4f28-a1db-d3965b103ba8.png">

kiaraarose replied:
I see. Good to know. Thanks!!

kiaraarose commented on Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionContextProxyCocoa.mm:
```diff
@@ -0,0 +1,65 @@
+    });
+}
+
+
```
nit: can remove extra newline

b-weinstein committed:
Make sendToProcessesForEvent compile correctly when used.
https://bugs.webkit.org/show_bug.cgi?id=249514
<rdar://problem/102820594>

Reviewed by Timothy Hatcher.

* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::sendToProcessesForEvent):

Fire the onBeforeNavigateEvent when a frame starts to load content.
https://bugs.webkit.org/show_bug.cgi?id=249514
rdar://102820594

Reviewed by Timothy Hatcher.

When a frame begins navigation, WebKit tells all extensions about it using the following flow.
- WebFrameLoaderClient sends a message to the WebExtensionController using IPC.
- WebExtensionController iterates over all the WebExtensionContexts and:
	- Sends a message to their WebExtensionContextProxy using IPC.

The WebExtensionContextProxy then handles this call by firing the onBeforeNavigate event on the extension.

There are some loose ends here:
1) We are firing events for some pages we shouldn't
2) The tab and frame IDs that we are vending to the listener aren't correct (they are identifiers, but not ones that have been
normalized for the required context)
3) We should be passing more information in the details dictionary.

* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm:
(WebKit::WebExtensionController::didStartProvisionalLoadForFrame): Tell every extension context about the event.
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionController.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionController.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionContextProxyCocoa.mm: Copied from Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.messages.in.
(WebKit::WebExtensionContextProxy::dispatchWebNavigationOnBeforeNavigateEvent): Create a details dictionary and
fire the "onBeforeNavigate" event.
* Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionControllerProxyCocoa.mm:
(WebKit::WebExtensionControllerProxy::didStartProvisionalLoadForFrame): Tell the UIProcess about this.
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.h:
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.messages.in:
* Source/WebKit/WebProcess/Extensions/WebExtensionControllerProxy.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDidStartProvisionalLoad): Tell the extensions controller about this.

Canonical link: https://commits.webkit.org/258163@main
